common/protob/messages-management.proto

```diff
@@ -427,28 +441,34 @@ message EntropyAck {
  * @next WordRequest
  */
 message RecoveryDevice {
-    optional uint32 word_count = 1;                     // number of words in BIP-39 mnemonic
-    optional bool passphrase_protection = 2;            // enable master node encryption using passphrase
-    optional bool pin_protection = 3;                   // enable PIN protection
-    optional string language = 4 [deprecated=true];     // deprecated (use ChangeLanguage)
-    optional string label = 5;                          // device label
-    optional bool enforce_wordlist = 6;                 // enforce BIP-39 wordlist during the process
-    // 7 reserved for unused recovery method
-    optional RecoveryDeviceType type = 8;               // supported recovery type
-    optional uint32 u2f_counter = 9;                    // U2F counter
-    optional bool dry_run = 10;                         // perform dry-run recovery workflow (for safe mnemonic validation)
+    optional uint32 word_count = 1;                           // number of words in BIP-39 mnemonic (T1 only)
+    optional bool passphrase_protection = 2;                  // enable master node encryption using passphrase
+    optional bool pin_protection = 3;                         // enable PIN protection
+    optional string language = 4 [deprecated=true];           // deprecated (use ChangeLanguage)
+    optional string label = 5;                                // device label
+    optional bool enforce_wordlist = 6;                       // enforce BIP-39 wordlist during the process (T1 only)
+    reserved 7;                                               // unused recovery method
+    optional RecoveryDeviceInputMethod input_method = 8;      // supported recovery input method (T1 only)
+    optional uint32 u2f_counter = 9;                          // U2F counter
+    optional RecoveryType type = 10 [default=NormalRecovery]; // the type of recovery to perform
     /**
      * Type of recovery procedure. These should be used as bitmask, e.g.,
-     * `RecoveryDeviceType_ScrambledWords | RecoveryDeviceType_Matrix`
+     * `RecoveryDeviceInputMethod_ScrambledWords | RecoveryDeviceInputMethod_Matrix`
      * listing every method supported by the host computer.
      *
      * Note that ScrambledWords must be supported by every implementation
      * for backward compatibility; there is no way to not support it.
      */
-    enum RecoveryDeviceType {
+    enum RecoveryDeviceInputMethod {
         // use powers of two when extending this field
-        RecoveryDeviceType_ScrambledWords = 0;        // words in scrambled order
-        RecoveryDeviceType_Matrix = 1;                // matrix recovery type
+        ScrambledWords = 0;        // words in scrambled order
+        Matrix = 1;                // matrix recovery type
+    }
+
+    enum RecoveryType {
```

Comment:
I see that `RecoveryType` is reused in `Features`. When a message or an enum is used in multiple messages, we have always placed it at the "global" level. I am not saying that it's wrong to embed it in `RecoveryDevice`, but rather unconventional.

Reply:
https://github.com/trezor/trezor-firmware/pull/3743/commits/d67569e737b91b66daa7d7ea6b84d25c40e13d26#diff-f014d9246998ed95f3b2fcc4df4ecb494f6b0652938fa4b83f91bf4194811a97R469